### PR TITLE
Add typings for Facebook Instant Games

### PR DESCRIPTION
**Pull request by menushka**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/fbinstant/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
+// Type definitions for FBInstant 6.1
```
Comment (minestarks) [inline types/fbinstant/index.d.ts]: `fbinstant` appears to be an unrelated npm package name. To avoid name collisions and to follow conventions set by other typings, could you rename to `facebook-instant-games`?

File: types/fbinstant/fbinstant-tests.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+declare const FBInstant: FBInstant;
+
+export class FBInstantTest {
```
Comment (minestarks) [inline types/fbinstant/fbinstant-tests.ts]: Removing this `export` keyword should be OK. You're not actually consuming this API as a module.

File: types/fbinstant/tslint.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        // Needs to be declares as it cannot be imported in typescript
+        // using the standard import statement
+        "strict-export-declare-modifiers": false
```
Comment (minestarks) [inline types/fbinstant/tslint.json]: Please remove this line. After you apply my suggested changes, do you still get errors?

File: types/fbinstant/fbinstant-tests.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+declare const FBInstant: FBInstant;
```
Comment (minestarks) [inline types/fbinstant/fbinstant-tests.ts]: Remove this line here, since the tests (or other users of these typings) shouldn't have to add any extra `declare` statements to get the typings to work.

If you change `index.d.ts` to the following, I think things will work as you intended:

```ts
declare namespace FBInstant {
    // put your variables here inside the namespace... 
    let player: Player;
    let context: Context;
    let payments: Payments;

    // ...

    // ... put your functions here inside the same namespace... 
    function getLocale(): string;
    function getPlatform(): string;
    function getSDKVersion(): string;

    // ...

    // .... add your types here in the same namespace ... 
    interface Player {
        getID(): string;
        getSignedPlayerInfoAsync(requestPayload: string): Promise<SignedPlayerInfo>;
   }

   // ...

   // ... add the other types here as well
   type LocalizationsDict = any;
   type SignedPurchaseRequest = string;

   // ...
}
```

Note that I'm using `interface` not class which I think reflects the actual API more accurately.

**Review by minestarks — CHANGES_REQUESTED** [review]

File: types/fbinstant/index.d.ts (hunk 1 above)
Comment (menushka) [reply types/fbinstant/index.d.ts]: Just to be sure, i should rename the folder in the types directory as well right?

Comment (minestarks) [reply types/fbinstant/index.d.ts]: Yep that's what I'm suggesting. Thanks!

**Comment by menushka** [comment]
Sorry for the sporadic commits, I'm using the API at work currently and now noticing some of the sparsely mentioned details.

**Review by minestarks — APPROVED** [review]
